staged-documentation.md

```diff
@@ -0,0 +1,282 @@
+    @Inject
+    private UserContext userContext;
+
+    @Override
+    public void acceptPageData(final PageData pageData) {
+        final ComponentBean component = createComponentBean();
+        pageData.getContent().addComponent(component);
+    }
+
+    private ComponentBean createComponentBean() {
+        final ComponentBean component = new ComponentBean();
+        //the template is in conf/templates/components/summercampaign/banner.hbs or in your theme project
+        component.setTemplateName("components/summercampaign/banner");//without .hbs!!!
+        Map<String, Object> data = new HashMap<>();
+        data.put("src", configuration.getString("summercampaign.img.src"));
+        data.put("width", configuration.getString("summercampaign.img.width"));
+        data.put("height", configuration.getString("summercampaign.img.height"));
+        data.put("href", demo.productcatalog.routes.SummerCampaignController.show(userContext.languageTag()).url());
+        component.setComponentData(data);
+        return component;
+    }
+}
+```
+
+You still need to wire the `ControllerComponent` to a controller or multiple controllers.
+Since it make sense to put it to all controllers except for checkout, we need a Guice Module:
+
+```java
+import com.commercetools.sunrise.framework.MultiControllerComponentResolver;
+import com.commercetools.sunrise.framework.MultiControllerComponentResolverBuilder;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import demo.productcatalog.SummerCampaignControllerComponent;
+
+public class ComponentsModule extends AbstractModule {
+    @Override
+    protected void configure() {
+    }
+
+    @Provides
+    public MultiControllerComponentResolver multiControllerComponentResolver() {
+        return new MultiControllerComponentResolverBuilder()
+                .add(SummerCampaignControllerComponent.class, controller -> !controller.getFrameworkTags().contains("checkout"))
+                .build();
+    }
+}
+```
+
+The `ComponentsModule` can be reused for multiple components, just add a `add` line for building the `MultiControllerComponentResolver`.
+
+Don't forget to register the module in `application.conf` with
+
+```
+play.modules.enabled += "absolute.path.to.your.ComponentsModule"
+```
+
+For `conf/templates/components/summercampaign/banner.hbs` we can use
+
+```handlebars
+<div class="row text-center">
+    <a href="{{componentData.href}}">
+        <p class="home-suggestions-title">Summer is here - buy sunglasses</p>
+        <img src="{{componentData.src}}" width="{{componentData.width}}" height="{{componentData.height}}" />
+    </a>
+</div>
+<hr class="home-suggestions-hr" />
+```
+
+And then it looks like
+![result](documentation-images/summercampaign-sungrasses-home.png)
+
+
+## Handlebars adding helpers
+
+*We suggest to not add handlebars helpers to use for calculations or logic, this is not what is handlebars designed for and selected for.*
```

Comment:
"We strongly suggest not to add Handlebars helpers to enable calculations or any kind of business logic in the templates. Handlebars helpers are only meant to simplify presentation logic (in contrast of a pure logic-less template such as [Mustache](https://mustache.github.io/)). Using helpers for other purposes would defeat the logic-less approach that we try to promote."